operators/src/util/raster_stream_to_geotiff.rs

```diff
@@ -248,43 +264,62 @@ impl<P: Pixel + GdalType> GdalDatasetWriter<P> {
 
         let tile_bounds = tile_info.spatial_partition();
 
-        let (upper_left, grid_array) = if self.output_bounds.contains(&tile_bounds) {
+        // compute the upper left pixel index in the output raster and extract the input data
+        let (output_ul_idx, grid_array) = if self.output_bounds.contains(&tile_bounds) {
             (
-                tile_bounds.upper_left(),
+                tile_info.global_upper_left_pixel_idx() - self.global_offset,
                 tile.into_materialized_tile().grid_array,
             )
         } else {
             // extract relevant data from tile (intersection with output_bounds)
 
-            let intersection = self
-                .output_bounds
-                .intersection(&tile_bounds)
-                .expect("tile must intersect with query");
-
-            let mut output_grid = MaskedGrid2D::from(EmptyGrid2D::new(intersection.grid_shape(
-                self.output_geo_transform.origin_coordinate,
-                self.output_geo_transform.spatial_resolution(),
-            )));
-
-            let offset = tile
-                .tile_geo_transform()
-                .coordinate_to_grid_idx_2d(intersection.upper_left());
-
-            let shifted_source = tile.grid_array.shift_by_offset(GridIdx([-1, -1]) * offset);
+            // TODO: compute the intersection on the `SpatialPartition2D`s once the float precision issue is fixed
+
+            let tile_offset = tile_info.global_upper_left_pixel_idx();
+            let tile_limit = tile_offset
+                + GridIdx2D::from([
+                    tile_info.tile_size_in_pixels.axis_size_y() as isize,
+                    tile_info.tile_size_in_pixels.axis_size_x() as isize,
+                ]);
+
+            if tile_limit.inner()[0] < self.global_offset.inner()[0]
```

Comment:
you can desstructure it with `GridIdx([idx_y, idx_x]) = ...` to make it more easy to unterstand.

Reply:
I destructured all the things b0d17f67c2c79b8581d6ed3c9d6c4b2cd66fa10e